src/mlpack/core/data/save_impl.hpp

```diff
@@ -105,11 +105,11 @@ bool Save(const std::string& filename,
 
 #ifdef ARMA_USE_HDF5
     // We can't save with streams for HDF5.
-    const bool success = (saveType == arma::hdf5_binary) ?
-        tmp.quiet_save(filename, saveType) :
-        tmp.quiet_save(stream, saveType);
+    const bool success = (saveType == file_type::mlp_hdf5_binary) ?
+        tmp.quiet_save(filename, ToArmaFileType(saveType)) :
+        tmp.quiet_save(stream, ToArmaFileType(saveType));
```

Comment:
This code also is only necessary if `MatType` is an Armadillo type. :+1:

Reply:
Yea, once fixed on a way to handle this case, I will make necessary changes everywhere...